runtime/handler.go

```diff
@@ -168,18 +168,30 @@ func handleForwardResponseOptions(ctx context.Context, w http.ResponseWriter, re
 	return nil
 }
 
-func handleForwardResponseStreamError(wroteHeader bool, marshaler Marshaler, w http.ResponseWriter, err error) {
-	buf, merr := marshaler.Marshal(streamChunk(nil, err))
-	if merr != nil {
-		grpclog.Infof("Failed to marshal an error: %v", merr)
+func handleForwardResponseStreamError(ctx context.Context, wroteHeader bool, marshaler Marshaler, w http.ResponseWriter, req *http.Request, mux *ServeMux, err error) {
+	if !wroteHeader {
+		// if we haven't already written header, use normal error handler to render
+		HTTPError(ctx, mux, marshaler, w, req, err)
 		return
 	}
-	if !wroteHeader {
-		s, ok := status.FromError(err)
-		if !ok {
-			s = status.New(codes.Unknown, err.Error())
+
+	// otherwise, render the error as a stream chunk in the response
+	var chunk map[string]proto.Message
+	if mux.streamErrorHandler != nil {
+		serr := mux.streamErrorHandler(ctx, err)
+		if serr == nil {
+			// TODO: log error about misbehaving error handler?
+			chunk = streamChunk(nil, err)
+		} else {
+			chunk = map[string]proto.Message{"error": (*internal.StreamError)(serr)}
```

Comment:
Not sure if this was discussed in #584, but this still doesn't give the user the power to fully control the envelope, right? Is that intentional?

Reply:
I added a comment earlier to that issue about why I felt that was ultimately a bad idea: a custom envelope in the runtime would make it impossible for a protoc plugin to generate a correct swagger/OpenAPI spec for the service.

Reply:
Fair enough